Fix shuffling for shows

andrewrabert opened the pull request:
**Changes**
Fixes shuffle playback of shows.

andrewrabert committed:
Fix shuffling for shows

dmitrylyzo commented:
I wonder how it will find the first unplayed episode in the randomized queue.

thornbill commented on src/components/playback/playbackmanager.js:
```diff
@@ -1872,6 +1872,7 @@ class PlaybackManager {
                 promise = apiClient.getEpisodes(firstItem.SeriesId || firstItem.Id, {
                     IsVirtualUnaired: false,
                     IsMissing: false,
+                    SortBy: options.shuffle ? 'Random' : 'SortName',
```
If we get episodes in the correct order by default, it may be better to do this:

```suggestion
                    SortBy: options.shuffle ? 'Random' : undefined,
```

nvllsvm replied:
Changed.

This is reasonable and seems to work. The server should respond with an appropriate default sort order when it's unspecified.

andrewrabert committed:
Include season id in request when retrieving episodes
This also fixes shuffling an individual season (broken since at least 10.8.13, probably earlier)

dmitrylyzo commented on src/components/playback/playbackmanager.js:
```diff
@@ -1868,48 +1868,37 @@ class PlaybackManager {
                 promise = apiClient.getEpisodes(firstItem.SeriesId || firstItem.Id, {
                     IsVirtualUnaired: false,
                     IsMissing: false,
+                    SeasonId: isSeason ? firstItem.Id : undefined,
```
Iirc, the idea was to continue playing the series (by starting a single season), including the next seasons.
Currently, this limits queue to the one season.

nvllsvm replied:
I'd be in favor of adding an explicit option to start playing a series at a particular season. Something like a `Play From Here` in the drop down.
![image](https://github.com/jellyfin/jellyfin-web/assets/6550543/377049d5-d0f3-47cc-9618-4069824d0695)

I do not think it should be the default however, maybe a library-wide option instead? Automatically starting the next season doesn't always make sense when, for example, someone clicks play on a `Specials` season. The expectation as of 10.8.13 is that only the `Specials` season would play. 
![image](https://github.com/jellyfin/jellyfin-web/assets/6550543/2105779b-7641-43e1-8e5e-eaf44bab8e84)

Also doesn't make sense for shows with self-contained seasons.
![image](https://github.com/jellyfin/jellyfin-web/assets/6550543/0837e7e3-825d-421f-a320-0e737069ade5)

dmitrylyzo replied:
Related PR #4827

nvllsvm replied:
I didn't realize this, but there's already an option to `Play all from here` shown on individual episodes in individual seasons. That option could likely be added to show on the season's drop down as well - thus implementing the behavior you're talking about.
